### PR TITLE
Allow Examples table to include values not used in steps

### PR DESCRIPTION
**Pull request by hicksjduk**
See https://github.com/pytest-dev/pytest-bdd/issues/390 for a description of the error to which this is a fix.

### CODE REVIEW

**Comment by hicksjduk** [comment]
@youtux, it seems that you were the last person to touch this code. Would you be able to review this PR, or else how do I request a review given that I don't have write access to the repo?

**Comment by elchupanebrej** [comment]
@hicksjduk Could you please rename the pull request to something like "Allow store values in Examples not used in steps" to show the real value of this feature

@youtux Please review

**Comment by hicksjduk** [comment]
Done.

File: pytest_bdd/parser.py [hunk 1]
```diff
@@ -283,10 +283,10 @@ def validate(self):
         """
         params = self.params
         example_params = self.get_example_params()
-        if params and example_params and params != example_params:
+        if params and example_params and any(p not in example_params for p in params):
```
Comment (youtux) [inline pytest_bdd/parser.py:286]: can we use set.issubset instead?

**Review by youtux — CHANGES_REQUESTED** [review]

File: pytest_bdd/parser.py (hunk 1 above)
Comment (hicksjduk) [reply pytest_bdd/parser.py]: Yes, of course. Change made.

**Review by olegpidsadnyi — COMMENTED** [review]
Cucumber (gherkin lint) uses no-unused-variables rule to optionally raise on this. @youtux Do you think we need to make it an option?

**Comment by hicksjduk** [comment]
> 
> 
> Cucumber (gherkin lint) uses no-unused-variables rule to optionally raise on this. @youtux Do you think we need to make it an option?

Isn't this a rule in the linter that checks the implementation code? Not in the functionality that implements the gherkin bindings. I can't find any reference in the cucumber documentation to this rule.

**Comment by elchupanebrej** [comment]
@hicksjduk Oleg means https://www.npmjs.com/package/gherkin-lint 
@olegpidsadnyi @hicksjduk please check #439 . User could explicitly decide to use this feature using new parameters in "scenario" and "scenarios"

**Comment by youtux** [comment]
I think we should just allow examples to include values that are not used. I wouldn't make it an option, I find it unnecessary complication.

**Comment by youtux** [comment]
@olegpidsadnyi please review

**Comment by jsa34** [comment]
@youtux  I don' think this restriction exists in the code base any more, from what I can see?

I can also see there is a test: _test_unused_params_ - is this covered by this test?

**Comment by youtux** [comment]
yes exactly this can be closed indeed

**Pull request closed without merging**